docker/puppetdb/docker-entrypoint.d/10-analytics.sh

```diff
@@ -1,28 +1,3 @@
 #!/bin/sh
 
-if [ "${PUPPERWARE_DISABLE_ANALYTICS}" = "true" ]; then
-    echo "($0) Pupperware analytics disabled; skipping metric submission"
-    exit 0
-fi
-
-# See: https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
-# Tracking ID
-tid=UA-132486246-1
-# Application Name
-an=puppetdb
-# Application Version
-av=$PUPPETDB_VERSION
-# Anonymous Client ID
-_file=/var/tmp/pwclientid
-cid=$(cat $_file 2>/dev/null || (cat /proc/sys/kernel/random/uuid | tee $_file))
-# Event Category
-ec=${PUPPERWARE_ANALYTICS_STREAM:-dev}
-# Event Action
-ea=start
-
-_params="v=1&t=event&tid=${tid}&an=${an}&av=${av}&cid=${cid}&ec=${ec}&ea=${ea}"
-_url="http://www.google-analytics.com/collect?${_params}"
-
-echo "($0) Sending metrics ${_url}"
-curl --fail --silent --show-error --output /dev/null \
-    -X POST -H "Content-Length: 0" $_url
+/analytics.sh start
```

Comment:
Hmmmmm... I'm not sure about having one script in `docker-entrypoint.d` and then another script outside, but I understand why you did it (although `stop` is not implemented). Seems like it might confuse us later on (and also don't like we have to add additional `COPY` steps in the `Dockerfile` rather than just using our established pattern).

Will have to think about this a bit more.

Reply:
Stop is implemented.

If we don't have a shared file at `/analytics.sh` then we'd have to have the exit script call into the entrypoint scripts, which is worse than this situation.

I think we're going to end up with some common files at `/` (or somewhere else, `/pupperware` or whatever) like we have with `/ssl.sh`.

That'd just be another pattern that we establish!

Good points though, no rush.

Reply:
Oh oh... I see that `stop` is just a value passed in the URI. I thought it was being used to alter behavior in the script - my mistake.

In any event, looks like we're going to close this per our discussion earlier anyhow!